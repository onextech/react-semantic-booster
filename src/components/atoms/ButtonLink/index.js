import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button, Icon } from 'semantic-ui-react';
import styled from 'styled-components';
import { countArrayLessUndefinedItems } from '@onextech/react-apollo-utils';


const hasManyChildren = props => countArrayLessUndefinedItems(props.children.props.children) > 1;
const Wrapper = styled(Button)`
  &.ui.button {
    padding: 0 !important;
    overflow: hidden;
    a {
      display: inline-block;  
      padding: ${props => (hasManyChildren(props) ? '.78571429em 1.5em' : '.78571429em')};
      color: inherit;
      &:hover, &:focus {
        text-decoration: none;
      }
      .icon {
        margin-right: ${props => (hasManyChildren(props) ? '.25rem' : '0')};
      }
    }
  }
`;

const ButtonLink = ({
  onClick,
  to,
  icon,
  content,
  ...rest
}) => (<Wrapper {...rest}>
  <Link to={to} onClick={onClick}>
    {icon && <Icon name={icon} />}
    {content}
  </Link>
</Wrapper>);

ButtonLink.propTypes = {
  to: PropTypes.string.isRequired,
  icon: PropTypes.string,
  content: PropTypes.string,
  onClick: PropTypes.func,
};

ButtonLink.defaultProps = {
  icon: undefined,
  content: undefined,
  onClick: undefined,
};

export default ButtonLink;
