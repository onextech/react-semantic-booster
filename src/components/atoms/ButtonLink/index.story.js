import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import ButtonLink from '.';

storiesOf('ButtonLink', module)
  .add('Default', () => (
    <ButtonLink
      content="Hello World"
      to="/"
      onClick={action('clicked')} />
  ))
  .add('Primary', () => (
    <ButtonLink
      content="Hello World"
      primary
      size="huge"
      to="/"
      onClick={action('clicked')} />
  ));
