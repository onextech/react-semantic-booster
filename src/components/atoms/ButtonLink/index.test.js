import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';
import { MemoryRouter as Router, Link } from 'react-router-dom';
import ButtonLink from '.';


describe('<ButtonLink>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Router><ButtonLink to="/">Facebook</ButtonLink></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has link child', () => {
    const wrapper = shallow(<ButtonLink to="/" />);
    expect(wrapper.find(Link).length).toBe(1);
  });

  it('has semantic button class', () => {
    const wrapper = render(<Router><ButtonLink to="/" /></Router>);
    expect(wrapper.hasClass('ui')).toBe(true);
  });
});
