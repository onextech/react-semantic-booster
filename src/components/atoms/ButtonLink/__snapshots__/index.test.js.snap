// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLink> renders correctly 1`] = `
<button
  className="ui button sc-bdVaJa jhbbHP"
  disabled={undefined}
  onClick={[Function]}
  role="button"
  tabIndex={undefined}
>
  <a
    href="/"
    onClick={[Function]}
  />
</button>
`;
